lib/Dialect/SV/SVOps.cpp

```diff
@@ -27,6 +27,13 @@ bool sv::isExpression(Operation *op) {
          isa<sv::ReadInterfaceSignalOp>(op);
 }
 
+LogicalResult sv::verifyInProceduralRegion(Operation *op) {
+  if (op->getParentWithTrait<sv::ProceduralRegion>())
+    return success();
```

Comment:
Is it enough that any parent be procedural or only the direct parent?  (let's assume IfOp is procedural and ifdef is split into procedural and non-procedural variants).

Reply:
I believe the appropriate condition is that there exists any parent with a procedural region.

> ifdef is split into procedural and non-procedural variant
Does `ifdef` create both procedural and non-procedural regions?

Reply:
IfDefOp exists in a graph region and its body is a graph region.  IfDefProceduralOp exists in procedural regions and its body is a procedural region.

Reply:
In other words, yes, checking direct parents should be enough

Reply:
Ok, it makes sense. Thank you!

Reply:
In the test([sv-dialect.mlir](https://github.com/llvm/circt/blob/be85f3cd8c8e9835551e16a7e7f5bf16a2ab6a66/test/ExportVerilog/sv-dialect.mlir#L162)), there is a fragment like following:

```
sv.initial {
  sv.ifdef .. {
     sv.fwrite "Foo"
  }
}
```
I think this `sv.fwrite` is invalid if we only check direct parents. Is the test case illegal?

Reply:
Yes, the testcase is illegal, we should reject this since sv.ifdef needs to be in a graph region:

```
sv.initial {
  sv.ifdef .. {
  }
}
```

the testcase should be adjusted to use `sv.ifdef.procedural`

Reply:
Ok, thanks!